staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +207,35 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// EventHandlerCount return the number of actually registered
+	// event handlers.
+	EventHandlerCount() int
+
+	// IsStopped reports whether the informer has already been stopped.
+	// Adding event handlers to already stopped informers is not possible.
+	IsStopped() bool
+
+	// IsStarted reports whether the informer has already been started
+	IsStarted() bool
+}
+
+// ResourceEventHandlerHandle is a handle returned by the
+// registration methods of SharedInformers for a registered
+// ResourceEventHandler. It can be used later on to remove
+// a registration again.
+// This indirection is required, because the ResourceEventHandler
+// interface can be implemented by non-go-comparable handlers, which
+// could not be removed from a list anymore.
+type ResourceEventHandlerHandle struct {
+	listener *processorListener
+}
+
+// IsActive reports whether this registration is still active
+// meaning that the handler registered with this handle is
+// still registered.
+func (h *ResourceEventHandlerHandle) IsActive() bool {
+	return h.listener != nil
```

Comment:
This looks likely to be racy. Do we really need this method?

Reply:
removed again

Reply:
It appears not removed? :)

I expected implementing this to need to make a call to a SharedInformer, or for this to be a method of a SharedInformer which takes in a ResourceEventHandlerHandle. But it is best if it just doesn't exist at all (and it is safe to remove ResourceEventHandlerHandle multiple times).

Reply:
yup, sorry, you are right. I made it local now to be still usable for the test coding. Additionally the modification of its field (listener) is now inside the synchronized code for the shared informer.